SYCL/ESIMD/api/esimd_bit_ops.cpp

```diff
@@ -0,0 +1,146 @@
+//==------- esimd_bit_ops.cpp  - DPC++ ESIMD on-device test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Smoke test for esimd bit operations APIs.
+
+#include "../esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+using namespace sycl::ext::intel::experimental::esimd;
+
+
+struct bit_op {
+  enum { cbit, fbl, fbh, num_ops };
+};
+
+template <typename T, int N, int Op> struct test_id;
+
+template <typename T> struct char_to_int {
+  using type = typename std::conditional<
+      sizeof(T) == 1,
+      typename std::conditional<std::is_signed<T>::value, int, unsigned>::type,
+      T>::type;
+};
+
+// The main test routine.
+template <typename T, int N, int Op> bool test(queue q) {
+  const char *ops[bit_op::num_ops] = {"cbit", "fbl", "fbh"};
+  std::cout << "Testing op=" << ops[Op] << " T=" << typeid(T).name()
+            << ", N=" << N << "...\n";
+
+  T val_all_zero{ 0 };
+  T val_all_one{ static_cast<T>(~val_all_zero) };
+  T val_two_one{ static_cast<T>(T{1} << (sizeof(T) * 8 - 2) | 2) }; // 010...010
+
+  T vals[] = { val_all_zero, val_all_one, val_two_one };
+  constexpr size_t num_vals = sizeof(vals) / sizeof(vals[0]);
+
+  constexpr size_t size = N * num_vals;
+  unsigned int *A = sycl::malloc_shared<unsigned int>(size, q);
+
+  for (unsigned int i = 0; i < size; ++i) {
```

Comment:
```suggestion
  for (unsigned int i = 0; i < num_vals; ++i) {
```

Reply:
why? I want to initialize the entire array.

Reply:
actually, I should allocate fewer elements too, will update (remnant from the previous version of the test)